src/emulator/errordisplay.c

```diff
@@ -0,0 +1,669 @@
+    char* var_r3;
+    s32 var_r30;
+    s32 nCellWidth;
+    s32 i;
+    s32 temp_r27;
+    char* temp_r3_2;
+
+    nCellWidth = sFontHeader->cellWidth;
+    DEMOGetROMFontSize(&nSize, &nSpace);
+    nSize *= 0x10;
+    nSpace *= 0x10;
+    pStringInfo->unk10 = 0;
+    pStringInfo->unk0C = 0;
+
+    if (pStringInfo->eStringID != SID_NONE) {
+        pStringInfo->szString = tableGetString(sBufferErrorStrings, pStringInfo->eStringID);
+
+        if (pStringInfo->szString != NULL) {
+            var_r27 = pStringInfo->szString;
+            pStringInfo->nLines = 1;
+
+            while (*var_r27 != '\0') {
+                temp_r3_2 = OSGetFontWidth(var_r27, &widthOut);
+
+                if (*var_r27 == '\n') {
+                    pStringInfo->nLines++;
+                    *var_r27 = '\0';
+                }
+
+                var_r27 = temp_r3_2;
+            }
+
+            var_r3 = pStringInfo->szString;
+            temp_r27 = pStringInfo->nLines;
+
+            for (i = 0; i < temp_r27; i++) {
+                var_r30 = 0;
+                var_r31 = NULL;
+
+                while (*var_r3 != '\0') {
+                    if (*var_r3 == ' ' || *var_r3 == '\t') {
+                        var_r31 = var_r3;
+                    }
+
+                    var_r3 = OSGetFontWidth(var_r3, &widthOut);
+                    var_r30 = ((s32)(nSize * widthOut) / nCellWidth) + var_r30 + nSpace;
+
+                    if (((var_r30 + 15) / 16 > 0x230) && var_r31 != NULL) {
+                        *var_r31 = 0;
+                        var_r3 = var_r31 + 1;
+                        var_r30 = 0;
+                        var_r31 = NULL;
+                        pStringInfo->nLines++;
+                    }
+                }
+
+                var_r3++;
+            }
+
+            pStringInfo->unk10 = ((pStringInfo->nLines * ((sFontHeader->leading * nSize) / nCellWidth)) + 15) / 16;
+        }
+    }
+}
+
+static void fn_80063910(ErrorDisplay* pErrorDisplay) {
+    s32 i;
+
+    pErrorDisplay->unk3C = 0;
+
+    if (pErrorDisplay->message.pStringInfo != NULL) {
+        pErrorDisplay->message.nShiftY = pErrorDisplay->message.pStringInfo->unk10;
+    } else {
+        pErrorDisplay->message.nShiftY = 0;
+    }
+
+    if (pErrorDisplay->nAction > 0) {
+        pErrorDisplay->message.nShiftY += DEMOGetRFTextHeight("\n");
+    }
+
+    pErrorDisplay->unk3C = pErrorDisplay->message.nShiftY;
+
+    for (i = 0; i < pErrorDisplay->nAction; i++) {
+        if (pErrorDisplay->action[i].message.pStringInfo != NULL) {
+            pErrorDisplay->action[i].message.nShiftY = pErrorDisplay->action[i].message.pStringInfo->unk10;
+        } else {
+            pErrorDisplay->action[i].message.nShiftY = 0;
+        }
+
+        pErrorDisplay->unk3C += pErrorDisplay->action[i].message.nShiftY;
+    }
+}
+
+/**
+ * @brief Prints a message.
+ * @param pEDString Pointer to `EDString`.
+ * @param nHeight The Y-position of the message to print.
+ * @param arg2 Unknown.
```

Comment:
I think this is something like "printf format string argument", it seems only used by `Create %ld block(s) of free space`